go-controller/pkg/factory/factory.go

```diff
@@ -7,12 +7,23 @@ import (
 	informerfactory "k8s.io/client-go/informers"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/tools/cache"
+	kapisnetworking "k8s.io/api/networking/v1"
+	kapi "k8s.io/api/core/v1"
+	"github.com/sirupsen/logrus"
+	"sync"
 )
 
 // WatchFactory initializes and manages common kube watches
 type WatchFactory struct {
 	iFactory  informerfactory.SharedInformerFactory
 	informers map[string]cache.SharedIndexInformer
+	handlers  map[string]federatedHandler
+}
+
+type federatedHandler struct {
+	mainHandler	cache.ResourceEventHandler
+	handlerMap	map[string]cache.ResourceEventHandler
+	handlerMutex    *sync.Mutex
```

Comment:
Could just embed a sync.Mutex into the handler?  Doesn't look like we need the pointer bit either since we're not passing it around anywhere?

```
type federatedHandler struct {
    sync.Mutex
    ...
}
```

Reply:
Right. I started with having a map of mutex, realised that was unnecessary and then this is vestige of that attempt. Will remove and embed the interface directly. Thanks.